src/test/components/README.md

```diff
@@ -1,3 +1,11 @@
 # React component testing
 
-This area is still experimental.
+## Snapshot tests
+
+Snapshots provide an easy way to show how React components are rendered. Any time a component is updated, its snapshot needs to be regenerated. This can be done by running `yarn test -u`. The updated snapshot should be manually inspected by both the author and code reviewer to ensure that the changes that were made are intentional and correct. Care should be taken when writing snapshot tests, as they fail quite easily, and only show that *something* has changed. Generally one snapshot test for a component is pretty good, while the larger component behavior should be asserted using more specific Enzyme tests.
+
+## Enzyme
+
+[Enzyme](https://github.com/airbnb/enzyme) tests are good for asserting specific behavior of components. They should be used for things like testing event handlers, and specific behavior of components. Originally this project used [react-test-renderer](https://www.npmjs.com/package/react-test-renderer), but Enzyme proved more useful for triggering events and manipulating the component as a user would with a mouse and keyboard.
```

Comment:
I'd like some guidance of how to use enzyme; because enzyme is very powerful it's easy to use it in a bad way (eg: manipulating props), and I think we should take great care to test the components from the point of view of the user.

Reply:
Thanks, good idea. I filed a follow-up: #1413